packages/core/docs/reference/plugins/amqp.md

```diff
@@ -1,295 +1,93 @@
 
-##### Publish and wait
-Method `publishAndWait()` sends a message via routing key, as [`publish()`](#publish).
+These properties and methods are available after Plugin initialization under the `service` namespace:
 
-###### Syntax
-```js
-amqp.publishAndWait(routingKey, message, publishOptions, [parentSpan])
-    .then((response) => {
-      // do whatever you want
-    })
-    .catch((err) => {
-      // either failed to send or response contained an error - work with it here
-    })
-```
+| Method | Description |
+|--------|-------------|
+| `amqp` | The Instance of the `@microfleet/transport-amqp` package. |
+| `amqp.publish(routingKey:string, message:any, [opts:object], [parentSpan])` | Publish message to specified route. |
+| `amqp.publishAndWait(routingKey:string, message:any, [opts:object], [parentSpan])` | Publish message to specified route and wait for response. |
+| `amqp.send(queueName:string, message:any, [publishOptions:object], [parentSpan])` | Publish message to specified queue directly. |
+| `amqp.sendAndWait(queueName:string, message:any, [publishOptions:object], [parentSpan])` | Publish message to specified queue directly and wait for response.  |
 
-###### Parameters
-  `routingKey` – Routing key (string)
-  `message` – Message to send (mixed)
-  `publishOptions` –  [Publish options](./amqp/publish-options.md) (object)
-  `parentSpan` – Parent span (object, optional)
+Please see [`@microfleet/transport-amqp`](https://github.com/microfleet/transport-amqp) to find the list of all available methods.
 
-###### Return value
-A Promise, which could be resolved with `undefined`, or rejected with an error. Resolving value depends on the 
-publish option `simpleResponse`.
+### `publishOptions` Object description
 
-##### Send `amqp.send(queueName, message, publishOptions, [parentSpan])`
-Method `send()` allows you to send a message to a queue directly. Otherwise it works similar to [`publish()`](#publish).
+The `publishOptions` parameter is a mix of `@microfleet/transport-amqp` internal options and `@microfleet/amqp-coffee` package options.
+Here's the list of some general options:
+| Option | Type | Default Value | Description |
+|--------|------|---------------|-------------|
+| `confirm` | `boolean` | `false` | Should plugin wait for confirmation before resolving? |
+| `immediate` | `boolean` |  `false` | Waits for the message to be delivered and resolves if it can, rejects otherwise, not implemented by RabbitMQ. |
+| `mandatory` | `boolean` | `false` | When true and message can't be routed to a queue – exception returned, otherwise its dropped. |
+| `contentType` | `string` | `application/json` | Default content-type for messages. |
+| `contentEncoding` | `string`| `plain` | Default content-encoding. |
+| `headers` | `object` | `{}` | Headers set. |
+| `simpleResponse` | `boolean` | `true` | Whether to return only response data or include headers etc. |
+| `deliveryMode` | `number` | `1` | `1` – transient, `2` – saved on disk. |
```

Comment:
Wait, as far as I remember it's snake cased `delivery_mode` when we pass it through `publishOptions` argument in amqp transport API, but in the [schema.js](https://github.com/microfleet/transport-amqp/blob/1b7fb18251e9aacfd0c8eedf733419ed7d399cdd/src/schema.js#L283) it's camel cased. It seems like it might be a bug either in the transport-amqp library or in the consuming code.

Reply:
Finally, we agreed that docs are right 👍 it's consuming code which has bugs